src/common/components/useVoiceCapabilities.ts

```diff
@@ -0,0 +1,74 @@
+import { getTTSEngine } from 'src/apps/chat/store-app-chat';
+import { CapabilitySpeechSynthesis } from '~/common/components/useCapabilities';
+
+import { useCapability as useElevenlabsCapability } from '~/modules/elevenlabs/elevenlabs.client'
+import { speakText as elevenlabsSpeakText } from '~/modules/elevenlabs/elevenlabs.client'
+import { EXPERIMENTAL_speakTextStream as EXPERIMENTAL_elevenlabsSpeakTextStream } from '~/modules/elevenlabs/elevenlabs.client'
+
+import { useCapability as useBrowserSpeechSynthesisCapability } from '~/modules/browser/speech-synthesis/browser.speechSynthesis.client'
+import { speakText as browserSpeechSynthesisSpeakText } from '~/modules/browser/speech-synthesis/browser.speechSynthesis.client'
+import { EXPERIMENTAL_speakTextStream as EXPERIMENTAL_browserSpeechSynthesisSpeakTextStream } from '~/modules/browser/speech-synthesis/browser.speechSynthesis.client'
+
+import { useElevenLabsVoices } from '~/modules/elevenlabs/useElevenLabsVoiceDropdown';
+import { useBrowserSpeechVoices } from '~/modules/browser/speech-synthesis/useBrowserSpeechVoiceDropdown';
+
+export const TTSEngineList: string[] = [
```

Comment:
See other note on this. I wonder if this would be better served as a map:
```typescript
export type TTSEngineKey = 'elevenlabs' | 'webspeech';

const TTSEngineList: { [key in TTSEngineKey]: { label: string, priority: number } } = {
  'elevenlabs': {
    label: 'ElevenLabs',
    priority: 2,
  },
  'webspeech': {
    label: 'Web Speech API',
    priority: 1,
  },
};
```
I've added an attribute called 'priority' to show how we can extend it in the future, for instance with default configurations.

Reply:
After a try. I think use map for extending is a good idea, but i suggest use a list contained map:

```ts
export type TTSEngineKey = 'Elevenlabs' | 'Web Speech API';
export type ASREngineKey = 'Web Speech API';

export const TTSEngineList: { key: TTSEngineKey; label: string }[] = [
  {
    key: 'Elevenlabs',
    label: 'ElevenLabs',
  },
  {
    key: 'Web Speech API',
    label: 'Web Speech API',
  },
];
```

because there are only 2 case using the variable now, one is `<FormRadioControl>` in `VoiceSettings.tsx` another one is `store-app-chat.ts`. `<FormRadioControl>` need Iterating and `VoiceSettings.tsx` need a default value. These are map can't do.